### PR TITLE
ui: make logout redirect to same auth method in login form

### PR DESCRIPTION
**Pull request by stefansundin**
Make logging out of Web UI redirect to the login form using the same auth method that was previously used. This makes it less annoying to log back in again when your session expires.

I was messing around and really did not think this would just work. I managed to get the dev server working and it just worked. I am sure you all want may want to make some changes to this, so let me know what should change. Also feel free to make changes in this branch, since maintainers have write access.

Thanks!!

### CODE REVIEW

**Comment by Monkeychip** [comment]
@stefansundin, your fix looks good, thank you for submitting the PR.  However, I wanted to check first and see if you knew about the similar functionality that already exists.  Under each auth method configuration, you have the option to "List method when unauthenticated".  If you select this, that method will then become the default method when logging in.  It's not the exact same solution, but would this solve your problem?



Here is a screenshot after having hard refreshed after logging out, with the OIDC method being marked as "List method when unauthenticated".

File: ui/app/routes/vault/cluster/logout.js [hunk 1]
```diff
@@ -16,13 +16,14 @@ export default Route.extend(ModelBoundaryRoute, {
   }),
 
   beforeModel() {
+    const authType = this.auth.getAuthType();
```
Comment (Monkeychip) [inline ui/app/routes/vault/cluster/logout.js:19]: change to `let` instead of `const`.

**Comment by stefansundin** [comment]
Interesting.. I had no idea what that checkbox did.

I don't have administrator privileges to this specific instance, but I can ask if they can check that checkbox.

I guess it's up to you whether or not to merge this. I think it may still be useful.

**Comment by Monkeychip** [comment]
@stefansundin We chatted it over and can't think of a reason why this wouldn't be an appreciated change.  I updated it with the master branch and once the tests come back clear, I'll go ahead and merge.  Thanks again for your work on this!

**Comment by stefansundin** [comment]
Wonderful! I am very happy to have made my first contribution to the vault project, even if it is very small. :)

**Pull request merged**